tests/test_minimal/test_metadata_schema.py

```diff
@@ -0,0 +1,53 @@
+import unittest
+from pathlib import Path
+
+import jsonschema
+
+from neuroconv.utils import load_dict_from_file
+from neuroconv.utils.json_schema import validate_metadata
+
+
+def test_metadata_schema():
```

Comment:
So, ideally, to really test that this schema catches violations properly, can you add a second test case that causes the error to be raised?

I'd use `pytest.raises` similar to [this example](https://github.com/catalystneuro/neuroconv/blob/554e07b96072eb6180eccfe3bdc4d049794a4f8d/tests/test_minimal/test_tools/test_backend_and_dataset_configuration/test_common_dataset_io_configuration_model.py#L63-L68)

Reply:
I opted to use the `iter_errors` method of the `jsonschema.Draft7Validator` class to get a list of all the errors for a given instance. 

I had to update the schema to have `additionalProperties: false` to get the `patternProperties` regex to actually factor into the validation. 

Currently, the schema allows for a plane without any additional nested dictionaries:
```json
{
"DfOverF": "{
             "name":"DfOverF",
             "PlaneName": {}
       },
}
```

Is this intended?

Reply:
It's technically allowed. The `DfOverF` is actually a container of `RoiResponseSeries`. And it's possible to have an empty container

But TBH I'd disallow it if that's easy (`minItems=1` perhaps?)